### PR TITLE
Azure Optimization Engine August release

### LINKED ISSUE #831: Underutilized Premium SSD disk recommendation not reporting results
## 🐛 Problem

The "Premium SSD disk has been underutilized" recommendation is not reporting results in situations where low usage disk metrics observed over the past 7 days would justify a recommendation.

## 👣 Repro steps

Provision a virtual machine with premium disks and let it run for several days without I/O load on premium disks (keeping it below 5% both for IOPS and throughput). One week later, AOE is not reporting any disk right-size recommendation for those disks.

## 🤔 Expected

The "Premium SSD disk has been underutilized" recommendation should report underutilized disks that have been exhibiting a usage below 5% both for IOPS and throughput.

### PR DESCRIPTION
**Pull request by helderpinto**
## 🛠️ Description

➕ Added:
1. `Register-MultitenantAutomationSchedules` PowerShell script helper to add a different Azure tenant to the scope of AOE.
2. ZRS disks included in the scope of the `Premium SSD disk has been underutilized` recommendation (besides LRS).
3. Option to scope consumption exports to MCA Billing Profile.

✏️ Changed:
1. Improved SQL Database security, replacing SQL authentication by Entra ID authentication-only.

🛠️ Fixed:
1. `Premium SSD disk has been underutilized` recommendation was not showing results due to a meter name change in Cost Management.
2. Consumption exports for Pay-As-You-Go MCA subscriptions were missing cost data.

Fixes #831 #828

### 🔬 How did you test this change?

> - [ ] 🤏 Lint tests
> - [ ] 🤞 PS -WhatIf / az validate
> - [X ] 👍 Manually deployed + verified
> - [ ] 💪 Unit tests
> - [X] 🙌 Integration tests

### 🙋‍♀️ Do any of the following that apply?

> - [ ] 🚨 This is a breaking change.
> - [ ] 🤏 The change is less than 20 lines of code.

### 📑 Did you update `docs/changelog.md`?

> - [X] ✅ Updated changelog (required for `dev` PRs)
> - [ ] ➡️ Will add log in a future PR (feature branch PRs only)
> - [ ] ❎ Log not needed (small/internal change)

### 📖 Did you update documentation?

> - [X] ✅ Public docs in `docs` (required for `dev`)
> - [ ] ✅ Internal dev docs in `src` (required for `dev`)
> - [ ] ➡️ Will add docs in a future PR (feature branch PRs only)
> - [ ] ❎ Docs not needed (small/internal change)

### CODE REVIEW

**Review by arthurclares — APPROVED** [review]

File: docs/README.md [hunk 1]
```diff
@@ -126,7 +126,7 @@ All the main changes are tracked in the changelog. For additional details, refer
...
-      <td align="center" valign="top" width="16.66%"><a href="https://aka.ms/helderpinto"><img src="https://avatars.githubusercontent.com/u/10661605?v=4?s=100" width="100px;" alt="Hélder Pinto"/><br /><sub><b>Hélder Pinto</b></sub></a><br /><a href="https://github.com/microsoft/finops-toolkit/commits?author=helderpinto" title="Documentation">📖</a> <a href="https://github.com/microsoft/finops-toolkit/issues?q=author%3Ahelderpinto" title="Bug reports">🐛</a></td>
+      <td align="center" valign="top" width="16.66%"><a href="https://aka.ms/helderpinto"><img src="https://avatars.githubusercontent.com/u/10661605?v=4?s=100" width="100px;" alt="Hélder Pinto"/><br /><sub><b>Hélder Pinto</b></sub></a><br /><a href="https://github.com/microsoft/finops-toolkit/commits?author=helderpinto" title="Code">💻</a> <a href="https://github.com/microsoft/finops-toolkit/pulls?q=is%3Apr+reviewed-by%3Ahelderpinto" title="Reviewed Pull Requests">👀</a> <a href="https://github.com/microsoft/finops-toolkit/commits?author=helderpinto" title="Documentation">📖</a> <a href="https://github.com/microsoft/finops-toolkit/issues?q=author%3Ahelderpinto" title="Bug reports">🐛</a></td>
```
Comment (arthurclares) [inline docs/README.md:129]: @flanakin Can you just double check if that is okay?

File: .all-contributorsrc [hunk 2]
```diff
@@ -159,6 +159,8 @@
       "avatar_url": "https://avatars.githubusercontent.com/u/10661605?v=4",
       "profile": "https://aka.ms/helderpinto",
       "contributions": [
+        "code",
```
Comment (arthurclares) [inline .all-contributorsrc:162]: @flanakin And also this one

**Review by arthurclares — APPROVED** [review]

File: .all-contributorsrc (hunk 2 above)
Comment (helderpinto) [reply .all-contributorsrc:162]: @flanakin, I didn't use any local NPM command. I didn't find any docs describing how these entries are expected to be updated. That's why I added them manually.

File: src/optimization-engine/runbooks/data-collection/Export-ConsumptionToBlobStorage.ps1 [hunk 3]
```diff
@@ -667,59 +675,128 @@ else
                             {
                                 $tags = $null
                             }
+
+                            if ([string]::IsNullOrEmpty($consumptionLine.properties.billingProfileId))
```
[the remark this answers is not in the document]
Comment (helderpinto) [reply src/optimization-engine/runbooks/data-collection/Export-ConsumptionToBlobStorage.ps1:679]: Yes, this API is still there for edge scenarios, such as MSDN subscriptions, which were not supported by the cost details API at the time of writing this code.

File: src/optimization-engine/runbooks/remediations/Remediate-LongDeallocatedVMsFiltered.ps1 [hunk 4]
```diff
@@ -86,7 +86,9 @@ $connectionSuccess = $false
 do {
     $tries++
     try {
-        $Conn = New-Object System.Data.SqlClient.SqlConnection("Server=tcp:$sqlserver,1433;Database=$sqldatabase;User ID=$SqlUsername;Password=$SqlPass;Trusted_Connection=False;Encrypt=True;Connection Timeout=$SqlTimeout;") 
+        $dbToken = Get-AzAccessToken -ResourceUrl "https://$azureSqlDomain/"
```
[the remark this answers is not in the document]
Comment (helderpinto) [reply src/optimization-engine/runbooks/remediations/Remediate-LongDeallocatedVMsFiltered.ps1:89]: Yes, it's absolutely awful :-) I never invested in modularizing this duplicate code. It's definitely something that will be prioritized in v2.

**Pull request merged**